exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentTraceExporterHandler.java

```diff
@@ -27,28 +27,31 @@
   private static final String DEFAULT_END_POINT = "localhost:55678";
   private static final String DEFAULT_SERVICE_NAME = "OpenCensus";
 
-  // private final String endPoint;
-  // private final Node node;
-  // private final boolean useInsecure;
+  @Nullable private final OcAgentTraceServiceClients.ExportServiceClient client;
 
   OcAgentTraceExporterHandler() {
     this(null, null, null);
   }
 
   OcAgentTraceExporterHandler(
       @Nullable String endPoint, @Nullable String serviceName, @Nullable Boolean useInsecure) {
-    // this.endPoint = endPoint == null ? DEFAULT_END_POINT : endPoint;
-    // if (serviceName == null) {
-    //   serviceName = DEFAULT_SERVICE_NAME;
-    // }
-    // this.node = OcAgentNodeUtils.getNodeInfo(serviceName);
-    // this.useInsecure = useInsecure == null ? false : useInsecure;
+    if (endPoint == null) {
+      endPoint = DEFAULT_END_POINT;
+    }
+    if (serviceName == null) {
+      serviceName = DEFAULT_SERVICE_NAME;
+    }
+    if (useInsecure == null) {
+      useInsecure = false;
+    }
+    client = OcAgentTraceServiceClients.openStreams(endPoint, useInsecure, serviceName);
```

Comment:
Probably I miss understand the current design, but I want to make sure we are on the same page. I think the whole idea was to watch for the oc-agent to run. So if we fail to connect we should retry every 1min (or configurable time). The main workflow is if somebody deploys this then after 1 week they decide that they need traces they will be able to just deploy the ocagent without restarting their app.

High-level design:
* In the application (or other library) we will have something like "registerOcAgentExporterWhenAvailable"
* Start a new Thread (for the purpose of these explanation say name = ThreadOcAgent) that tries to connect to the oc-agent every 1 minute.
* If not able to connect try again in 1 minute (or configurable time).
* If able to connect:
    * Install the OcAgentSpanExportHandler to the SpanExporter. So spans will be exported.
    * When disconnected uninstall the OcAgentSpanExportHandler from the SpanExporter, also ThreadOcAgent should again retry to connect in 1 minute.
    * The config rpc handling can be done in the ThreadOcAgent as well.

Reply:
Discussed offline, we think keep retrying is the right thing to do here. Will update the code based on that.

Reply:
Sorry I hadn't seen this ping, I'll also create an issue for the Go agent exporter.